src/azure-cli-core/azure/cli/core/_msal.py

```diff
@@ -19,16 +19,14 @@ def _acquire_token_silent_by_finding_rt_belongs_to_me_or_my_family(
 
     def _acquire_token_silent_by_finding_specific_refresh_token(
```

Comment:
We can possibly deprecate this whole file given MSAL already has `ClientApplication.acquire_token_by_refresh_token`. 

However, it doesn't take `kwargs`. Created PR https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/298 to fix it.

Reply:
MSAL has decided to support `kwargs` in `ClientApplication.acquire_token_by_refresh_token`. Let's put this PR on hold while the new MSAL is released.

Reply:
msal 1.9 was released two days ago.